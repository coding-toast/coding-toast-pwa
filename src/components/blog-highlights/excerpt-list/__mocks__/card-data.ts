export const query = [
  {
    blogId: '968769ef-231d-5501-8942-ee51d8d1324d',
    title: 'blog-post-placerholder1',
    publishDate: '`January 1, 1970`',
    description: 'first blog post',
    banner: 'https://cliparting.com/wp-content/uploads/2016/06/Ribbon-outline-banner-clip-art-free-vector-for-free-download-about.jpg'
  },
  {
    blogId: '1a172731-99f4-5f68-8ead-23324a44f70c',
    title: 'blog-post-placerholder3',
    publishDate: '`January 1, 1970`',
    description: 'third blog post',
    banner: 'https://cliparting.com/wp-content/uploads/2016/06/Ribbon-outline-banner-clip-art-free-vector-for-free-download-about.jpg'
  },
  {
    blogId: '011af6f6-6c3f-5eab-81ad-3f73e427dbd8',
    title: 'blog-post-placerholder4',
    publishDate: '`January 1, 1970`',
    description: 'fourth blog post',
    banner: 'https://cliparting.com/wp-content/uploads/2016/06/Ribbon-outline-banner-clip-art-free-vector-for-free-download-about.jpg'
  },
  {
    blogId: 'd28f2392-be5f-5fc1-96fb-b3c69e7dce6b',
    title: 'blog-post-placerholder2',
    publishDate: '`January 1, 1970`',
    description: 'second blog post',
    banner: 'https://cliparting.com/wp-content/uploads/2016/06/Ribbon-outline-banner-clip-art-free-vector-for-free-download-about.jpg'
  }
];
