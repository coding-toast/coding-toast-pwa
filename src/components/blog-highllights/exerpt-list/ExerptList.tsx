import * as React from 'react'
import { CardBody, CardTitle, CardDesctiption, CardMeta } from '../../card'

interface ExcerptListProps {
  allMarkdownRemark: {
    allMarkdownRemark: {
      edges: {
        node: {
          frontmatter: {
            author: string
            banner: string
            description: string
            title: string
            publishDate: string
          }
        }
      }[]
    }
  }
}

const ExerptList: React.FC<ExcerptListProps> = props => {
  const { allMarkdownRemark } = props

  return (
    <section>
      {allMarkdownRemark.allMarkdownRemark.edges.map((post: any) => (
        <CardBody>
          <CardTitle title={post.node.frontmatter.title} backgroundUrl={post.node.frontmatter.banner} />
          <CardDesctiption text={post.node.frontmatter.description} />
          <CardMeta publishDate={post.node.frontmatter.publishDate} />
        </CardBody>
      ))}
    </section>
  )
}

export default ExerptList
