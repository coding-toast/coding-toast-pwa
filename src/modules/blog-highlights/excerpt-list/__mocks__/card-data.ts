export const query = {
  allStrapiBlogs: {
    edges: [
      {
        node: {
          blogId: '968769ef-231d-5501-8942-ee51d8d1324d',
          title: 'blog-post-placerholder1',
          publishDate: '`January 1, 1970`',
          description: 'first blog post',
          banner: 'https://cliparting.com/wp-content/uploads/2016/06/Ribbon-outline-banner-clip-art-free-vector-for-free-download-about.jpg'
        }
      },
      {
        node: {
          blogId: '68e1f043-9989-584f-80e7-2bb2eed3f985',
          title: 'blog-post-placerholder5',
          publishDate: '`January 1, 1970`',
          description: 'fifth blog post',
          banner: 'https://cliparting.com/wp-content/uploads/2016/06/Ribbon-outline-banner-clip-art-free-vector-for-free-download-about.jpg'
        }
      },
      {
        node: {
          blogId: '1a172731-99f4-5f68-8ead-23324a44f70c',
          title: 'blog-post-placerholder3',
          publishDate: '`January 1, 1970`',
          description: 'third blog post',
          banner: 'https://cliparting.com/wp-content/uploads/2016/06/Ribbon-outline-banner-clip-art-free-vector-for-free-download-about.jpg'
        }
      },
      {
        node: {
          blogId: '011af6f6-6c3f-5eab-81ad-3f73e427dbd8',
          title: 'blog-post-placerholder4',
          publishDate: '`January 1, 1970`',
          description: 'fourth blog post',
          banner: 'https://cliparting.com/wp-content/uploads/2016/06/Ribbon-outline-banner-clip-art-free-vector-for-free-download-about.jpg'
        }
      },
      {
        node: {
          blogId: 'd28f2392-be5f-5fc1-96fb-b3c69e7dce6b',
          title: 'blog-post-placerholder2',
          publishDate: '`January 1, 1970`',
          description: 'second blog post',
          banner: 'https://cliparting.com/wp-content/uploads/2016/06/Ribbon-outline-banner-clip-art-free-vector-for-free-download-about.jpg'
        }
      }
    ]
  }
}
