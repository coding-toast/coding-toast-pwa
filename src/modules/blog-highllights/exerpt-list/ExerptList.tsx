import * as React from 'react'
import { CardBody, CardTitle, CardDesctiption, CardMeta } from '../../../components/card'
import ExerptListStyles from './ExerptList.module.scss'

interface ExcerptListProps {
  allMarkdownRemark: {
    allMarkdownRemark: {
      edges: Post[]
    }
  }
}
interface Post {
  node: {
    frontmatter: {
      author: string
      banner: string
      description: string
      title: string
      publishDate: string
    }
  }
}
const ExerptList: React.FC<ExcerptListProps> = props => {
  const { allMarkdownRemark } = props
  return (
    <section className={ExerptListStyles.list}>
      {allMarkdownRemark.allMarkdownRemark.edges.map((post: Post) => {
        const { title, banner, description, publishDate } = post.node.frontmatter
        return (
          <CardBody>
            <CardTitle title={title} backgroundUrl={banner} />
            <CardDesctiption text={description} />
            <CardMeta publishDate={publishDate} />
          </CardBody>
        )
      })}
    </section>
  )
}
export default ExerptList
