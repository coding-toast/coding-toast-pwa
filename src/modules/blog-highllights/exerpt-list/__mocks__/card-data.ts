export const query = {
  allMarkdownRemark: {
    edges: [
      {
        node: {
          frontmatter: {
            author: 'author here',
            banner:
              'https://cliparting.com/wp-content/uploads/2016/06/Ribbon-outline-banner-clip-art-free-vector-for-free-download-about.jpg',
            description: 'second blog post',
            title: 'blog-post-placerholder2',
            publishDate: 'June 16, 2007'
          }
        }
      },
      {
        node: {
          frontmatter: {
            author: 'author here',
            banner:
              'https://cliparting.com/wp-content/uploads/2016/06/Ribbon-outline-banner-clip-art-free-vector-for-free-download-about.jpg',
            description: 'third blog post',
            title: 'blog-post-placerholder3',
            publishDate: 'January 30, 2001'
          }
        }
      },
      {
        node: {
          frontmatter: {
            author: 'author here',
            banner:
              'https://cliparting.com/wp-content/uploads/2016/06/Ribbon-outline-banner-clip-art-free-vector-for-free-download-about.jpg',
            description: 'fourth blog post',
            title: 'blog-post-placerholder4',
            publishDate: 'February 13, 1995'
          }
        }
      },
      {
        node: {
          frontmatter: {
            author: 'author here',
            banner:
              'https://cliparting.com/wp-content/uploads/2016/06/Ribbon-outline-banner-clip-art-free-vector-for-free-download-about.jpg',
            description: 'fifth blog post',
            title: 'blog-post-placerholder5',
            publishDate: 'December 7, 2005'
          }
        }
      },
      {
        node: {
          frontmatter: {
            author: 'author here',
            banner:
              'https://cliparting.com/wp-content/uploads/2016/06/Ribbon-outline-banner-clip-art-free-vector-for-free-download-about.jpg',
            description: 'first blog post',
            title: 'blog-post-placerholder1',
            publishDate: 'December 29, 2015'
          }
        }
      }
    ]
  }
}
