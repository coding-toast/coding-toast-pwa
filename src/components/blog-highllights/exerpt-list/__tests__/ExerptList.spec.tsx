import React from 'react'
import renderer from 'react-test-renderer'
import ExerptList from '../ExerptList'
import { query } from '../__mocks__/card-data'

it('renders correctly', () => {
  const tree = renderer.create(<ExerptList allMarkdownRemark={query} />).toJSON()
  expect(tree).toMatchSnapshot()
})
