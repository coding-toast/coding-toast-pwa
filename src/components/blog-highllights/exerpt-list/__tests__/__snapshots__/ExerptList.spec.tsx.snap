// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<section>
  <article
    className="card"
  >
    <div
      style={
        Object {
          "backgroundImage": "url(https://cliparting.com/wp-content/uploads/2016/06/Ribbon-outline-banner-clip-art-free-vector-for-free-download-about.jpg)",
          "backgroundSize": "cover",
          "height": "124px",
          "position": "relative",
        }
      }
    >
      <h2
        style={
          Object {
            "bottom": "12px",
            "margin": 0,
            "padding": 0,
            "position": "absolute",
          }
        }
      >
        blog-post-placerholder2
      </h2>
    </div>
    <div>
      <p>
        second blog post
      </p>
    </div>
    <p>
      June 16, 2007
    </p>
  </article>
  <article
    className="card"
  >
    <div
      style={
        Object {
          "backgroundImage": "url(https://cliparting.com/wp-content/uploads/2016/06/Ribbon-outline-banner-clip-art-free-vector-for-free-download-about.jpg)",
          "backgroundSize": "cover",
          "height": "124px",
          "position": "relative",
        }
      }
    >
      <h2
        style={
          Object {
            "bottom": "12px",
            "margin": 0,
            "padding": 0,
            "position": "absolute",
          }
        }
      >
        blog-post-placerholder3
      </h2>
    </div>
    <div>
      <p>
        third blog post
      </p>
    </div>
    <p>
      January 30, 2001
    </p>
  </article>
  <article
    className="card"
  >
    <div
      style={
        Object {
          "backgroundImage": "url(https://cliparting.com/wp-content/uploads/2016/06/Ribbon-outline-banner-clip-art-free-vector-for-free-download-about.jpg)",
          "backgroundSize": "cover",
          "height": "124px",
          "position": "relative",
        }
      }
    >
      <h2
        style={
          Object {
            "bottom": "12px",
            "margin": 0,
            "padding": 0,
            "position": "absolute",
          }
        }
      >
        blog-post-placerholder4
      </h2>
    </div>
    <div>
      <p>
        fourth blog post
      </p>
    </div>
    <p>
      February 13, 1995
    </p>
  </article>
  <article
    className="card"
  >
    <div
      style={
        Object {
          "backgroundImage": "url(https://cliparting.com/wp-content/uploads/2016/06/Ribbon-outline-banner-clip-art-free-vector-for-free-download-about.jpg)",
          "backgroundSize": "cover",
          "height": "124px",
          "position": "relative",
        }
      }
    >
      <h2
        style={
          Object {
            "bottom": "12px",
            "margin": 0,
            "padding": 0,
            "position": "absolute",
          }
        }
      >
        blog-post-placerholder5
      </h2>
    </div>
    <div>
      <p>
        fifth blog post
      </p>
    </div>
    <p>
      December 7, 2005
    </p>
  </article>
  <article
    className="card"
  >
    <div
      style={
        Object {
          "backgroundImage": "url(https://cliparting.com/wp-content/uploads/2016/06/Ribbon-outline-banner-clip-art-free-vector-for-free-download-about.jpg)",
          "backgroundSize": "cover",
          "height": "124px",
          "position": "relative",
        }
      }
    >
      <h2
        style={
          Object {
            "bottom": "12px",
            "margin": 0,
            "padding": 0,
            "position": "absolute",
          }
        }
      >
        blog-post-placerholder1
      </h2>
    </div>
    <div>
      <p>
        first blog post
      </p>
    </div>
    <p>
      December 29, 2015
    </p>
  </article>
</section>
`;
